src/components/axes/timeAxis.ts

```diff
@@ -449,6 +449,26 @@ export module Axis {
         );
       };
 
+      var visibleTickMarks = this._tierMarkContainers[index]
+                                    .selectAll("." + AbstractAxis.TICK_MARK_CLASS)
+                                    .filter(function(d: any, i: number) {
+                                      return d3.select(this).style("visibility") === "visible";
+                                    });
+
+      // We use the ClientRects because x1/x2 attributes are not comparable to ClientRects of labels
+      var visibleTickMarkRects = visibleTickMarks[0].map((mark: any) => mark.getBoundingClientRect() );
```

Comment:
Try to avoid any wherever possible.  The only cases where we should have any's ideally is when dealing with raw data that we have no typing information for.